### PR TITLE
Enable prebuilt detection

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Fixes: https://github.com/dotnet/fsharp/issues/15166

This change enables prebuilt detection, for source-build.

After this is merged, I will add a subscription for https://github.com/dotnet/source-build-reference-packages

### CODE REVIEW

**Comment by NikolaMilosavljevic** [comment]
@MichaelSimons @mthalman @mmitche

**Comment by vzarytovskii** [comment]
Is there a reason helix getting removed? Is it implicit?

**Comment by vzarytovskii** [comment]
Also, probably a good idea to retarget it for 17.7 branch, since it's getting inserted everywhere currently (7.0.4xx and main)

**Comment by NikolaMilosavljevic** [comment]
> Is there a reason helix getting removed? Is it implicit?

It's part of Arcade toolset, so the explicit dependency isn't needed - @mmitche is this correct?

**Comment by NikolaMilosavljevic** [comment]
Hmm, after retargeting to `release/dev17.7` we are getting a bunch of new prebuilts - investigating...

**Comment by NikolaMilosavljevic** [comment]
> Hmm, after retargeting to `release/dev17.7` we are getting a bunch of new prebuilts - investigating...

Resolved all new issues.

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 1]
```diff
@@ -1,5 +1,16 @@
 <UsageData>
   <IgnorePatterns>
-    <UsagePattern IdentityGlob="*/*" />
+    <UsagePattern IdentityGlob="Microsoft.SourceBuild.Intermediate.*/*" />
+
+    <!-- Ignoring the known prebuilts. -->
+    
+    <!-- Tracked in https://github.com/dotnet/source-build/issues/3437 -->
+    <UsagePattern IdentityGlob="Microsoft.AspNetCore.App.Ref/7.0.2" />
```
Comment (MichaelSimons) [inline eng/SourceBuildPrebuiltBaseline.xml:8]: The 7.0.2 app ref related package should not be added to SBRP.  These are all from fsharp targetting net7.0.  It needs to target net8.0 for source build.  This is tracked with https://github.com/dotnet/fsharp/issues/14765

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 2]
```diff
@@ -4,10 +4,12 @@
 
     <!-- Ignoring the known prebuilts. -->
     
-    <!-- Tracked in https://github.com/dotnet/source-build/issues/3437 -->
+    <!-- These will go away when repo updates targeting to net8.0 -->
     <UsagePattern IdentityGlob="Microsoft.AspNetCore.App.Ref/7.0.2" />
     <UsagePattern IdentityGlob="Microsoft.NETCore.App.Host.linux-x64/7.0.2" />
     <UsagePattern IdentityGlob="Microsoft.NETCore.App.Ref/7.0.2" />
+
+    <!-- Tracked in https://github.com/dotnet/source-build/issues/3437 -->
```
Comment (MichaelSimons) [inline eng/SourceBuildPrebuiltBaseline.xml:12]: Thoughts on adding these two SBRP right away to avoid the need for a follow-up PR?

Comment (NikolaMilosavljevic) [reply eng/SourceBuildPrebuiltBaseline.xml:15]: Yeah - I wasn't sure if we want to add 7.0.2 SBRPs at all. Working on these two right now.

Comment (NikolaMilosavljevic) [reply eng/SourceBuildPrebuiltBaseline.xml:15]: `Microsoft.VisualStudio.Setup.Configuration.Interop/3.2.2146` is coming from MSBuild, dependency of `Microsoft.Build.Tasks.Core/17.7.0-preview-23217-02`.

When this repo updates MSBuild dependency, this package might get revved to a newer one, which will necessitate a new SBRP package.

Alternative fix would be to modify the package exclusion to use wildcard for version.

File: eng/Versions.props [hunk 3]
```diff
@@ -109,7 +109,6 @@
     <MicrosoftVisualStudioShellPackagesVersion>17.7.35338-preview.1</MicrosoftVisualStudioShellPackagesVersion>
     <VisualStudioProjectSystemPackagesVersion>17.7.58-pre</VisualStudioProjectSystemPackagesVersion>
     <MicrosoftVisualStudioThreadingPackagesVersion>17.7.5-preview</MicrosoftVisualStudioThreadingPackagesVersion>
-    <MicrosoftBuildOverallPackagesVersion>17.7.0-preview-23217-02</MicrosoftBuildOverallPackagesVersion>
     <!-- Roslyn packages -->
```
Comment (NikolaMilosavljevic) [inline eng/Versions.props:112]: I believe this needs to be fixed as well, to introduce proper `roslyn` flow into `fsharp`, and get rid of `RoslynVersion` property. We've made a similar change in `razor` repo: https://github.com/dotnet/razor/pull/8492

cc @mmitche @MichaelSimons

**Comment by vzarytovskii** [comment]
@NikolaMilosavljevic shall we merge it?

**Comment by NikolaMilosavljevic** [comment]
> @NikolaMilosavljevic shall we merge it?

@mmitche any concerns with the question I had above? https://github.com/dotnet/fsharp/pull/15209#discussion_r1192626579

**Comment by NikolaMilosavljevic** [comment]
> @NikolaMilosavljevic shall we merge it?

Thanks @vzarytovskii - this is ready to be merged - I've synced with @mmitche offline.

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**